From: Tim Chen <tim.c.chen@linux.intel.com>
Date: Wed, 15 Nov 2017 12:24:19 -0800
Subject: x86/idle: Disable IBRS when offlining a CPU and re-enable on wakeup
Patch-mainline: submitted on 2018/1/9
References: bsc#1068032

Clear IBRS when cpu is offlined and set it when brining it back online.

Signed-off-by: Tim Chen <tim.c.chen@linux.intel.com>
[ Switch to accessors. ]
Signed-off-by: Borislav Petkov <bp@suse.de>
---
 arch/x86/kernel/smpboot.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/arch/x86/kernel/smpboot.c
+++ b/arch/x86/kernel/smpboot.c
@@ -71,6 +71,7 @@
 
 #include <asm/smpboot_hooks.h>
 #include <asm/i8259.h>
+#include <asm/spec_ctrl.h>
 
 /* State of each CPU */
 DEFINE_PER_CPU(int, cpu_state) = { 0 };
@@ -1432,8 +1433,12 @@ void native_play_dead(void)
 	play_dead_common();
 	tboot_shutdown(TB_SHUTDOWN_WFS);
 
+	x86_disable_ibrs();
+
 	mwait_play_dead();	/* Only returns on failure */
 	hlt_play_dead();
+
+	x86_enable_ibrs();
 }
 
 #else /* ... !CONFIG_HOTPLUG_CPU */
